draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/TitleBlockZen.spec.tsx

```diff
@@ -642,6 +643,52 @@ describe("<TitleBlockZen />", () => {
     })
   })
 
+  describe("survey status", () => {
+    it("it doesn't render tag when field is omitted", async () => {
+      render(<TitleBlockZen title="Test Title">Example</TitleBlockZen>)
+
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("survey-status-tag")
+        ).not.toBeInTheDocument()
+      })
+    })
+
+    const surveyStatuses = ["draft", "live", "closed", "scheduled"]
+    const statusClassNameMap = {
+      draft: "statusDraft",
+      live: "statusLive",
+      closed: "statusClosed",
+      scheduled: "statusClosed",
+    }
+    surveyStatuses.forEach(status => {
+      it(`it renders tag with correct text and variant when ${status} status`, async () => {
+        render(
+          <TitleBlockZen
+            title="Test Title"
+            surveyStatus={{
+              text: `${status} text`,
+              status: `${status}` as
+                | "draft"
+                | "live"
+                | "scheduled"
+                | "closed"
+                | "default",
+            }}
+          >
+            Example
+          </TitleBlockZen>
+        )
+
+        const tagElement = await (
+          await screen.findByTestId("survey-status-tag")
+        ).firstChild
```

Comment:
To echo a comment of @lijuenc's - is the second `await` necessary? Does this mean that `.firstChild` is an asynchronous method?